server/src/App/Models/Event.php

```diff
@@ -43,6 +43,7 @@ public function __construct(array $attributes = [])
             'start_date'   => V::notEmpty()->date(),
             'end_date'     => V::callback([$this, 'checkEndDate']),
             'is_confirmed' => V::notOptional()->boolType(),
+            'is_closed'    => V::notOptional()->boolType(),
```

Comment:
Il faudrait sans doute ajouter une méthode de validation custom pour checker l'intégrité de ce champ.  
(Du genre: Ne pas pouvoir passer en `is_closed` si le champ `is_confirmed` est à `false` ou que `start_date` n'est pas passé)

_(cf. le champ `end_date` pour la façon de mettre en place une méthode de validation custom)_

Reply:
`'is_closed'    => V::callback([$this, 'checkClosedState'])`
```
public function checkClosedState($value)
{
    $boolChecker = V::notOptional()->boolType();
    if (!$boolChecker->validate($value)
    || !$boolChecker->validate($this->is_confirmed)) {
        return false;
    }

    if (!$value) {
        return true;
    }

    $endDate = new \DateTime($this->end_date);
    $now     = new \DateTime();
    $isPastAndConfirmed = $this->is_confirmed && ($endDate < $now);

    return $isPastAndConfirmed ?: 'eventCannotBeClosed';
}
```
Ça convient ?